operators/pkg/instance-controller/cloudinit.go

```diff
@@ -101,3 +117,40 @@ func (r *InstanceReconciler) getWebDavCredentials(ctx context.Context, secretNam
 
 	return string(userBytes), string(passBytes), nil
 }
+
+// GetPublicKeys extracts and returns the set of public keys associated with a
+// given tenant, along with the ones of the tenants having Manager role in the
+// corresponding workspace.
+func (r *InstanceReconciler) GetPublicKeys(ctx context.Context) ([]string, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	// Retrieve the public keys from the tenant owning the instance.
+	tenant := clctx.TenantFrom(ctx)
+	publicKeys := append(make([]string, 0), tenant.Spec.PublicKeys...)
+	log.V(utils.LogDebugLevel).Info("public keys correctly retrieved", "number", len(publicKeys))
+
+	// Retrieve the template associated with the instance to retrieve the name of the workspace.
+	template := clctx.TemplateFrom(ctx)
+	workspaceName := template.Spec.WorkspaceRef.Name
+	labelSelector := map[string]string{clv1alpha1.WorkspaceLabelPrefix + workspaceName: string(clv1alpha1.Manager)}
+
+	var managers clv1alpha1.TenantList
+	if err := r.List(ctx, &managers, client.MatchingLabels(labelSelector)); kerrors.IsNotFound(err) {
+		// if there are no managers in this workspace there is nothing to do
+		log.V(utils.LogDebugLevel).Info("no managers found for workspace", "workspace", workspaceName)
+		return publicKeys, nil
```

Comment:
Can the `.List` method return a notFound error? I actually found no documentation about this...

Reply:
Probably not